File: 0001-two-sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> map;

        for(int i=0; i < nums.size(); i++){
            int comp = target - nums[i];
            if(map.find(comp) != map.end()){
                return std::vector<int>{map[comp], i};
            }
            else map.insert(std::make_pair(nums[i], i));
        }
        return std::vector<int>{};
    }
};

File: 0001-two-sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        data = {}

        for i in range(0,len(nums)):
            comp = target - nums[i]
            if comp in data:
                return [data[comp], i]
            else: data[nums[i]] = i
        return []
